packages/synapse-react-client/src/components/ChallengeTeamWizard/ChallengeTeamWizard.tsx

```diff
@@ -79,366 +59,368 @@ export type ChallengeTeamWizardProps = {
+      enabled: Boolean(isLoggedIn && challenge),
+    })
+
+  const isMemberOfSubmissionTeam =
+    userSubmissionTeams && userSubmissionTeams.results.length > 0
+
+  const {
+    data: selectedTeamMembershipStatus,
+    isLoading: selectedTeamMembershipStatusIsLoading,
+    error: selectedTeamMembershipStatusError,
+  } = useGetMembershipStatus(selectedTeamId!, String(userProfile?.ownerId), {
+    enabled: isLoggedIn && !!selectedTeamId && !!userProfile,
+  })
+
+  const addUserToTeam = useCallback(async () => {
+    if (!selectedTeamId || !userProfile) return
+    await addTeamMember({
+      teamId: selectedTeamId,
+      userId: userProfile.ownerId,
+    })
+    setStep(ChallengeTeamWizardStep.REGISTRATION_SUCCESSFUL)
+  }, [addTeamMember, setStep, selectedTeamId, userProfile])
+
+  const hide = useCallback(() => {
+    setHasCreatedNewTeam(false)
+    setSelectedTeamId(undefined)
+    onClose()
+  }, [onClose])
+
+  const isLoading = isLoadingUserProfile || isLoadingChallenge
+
+  const backButton = useMemo(() => {
+    switch (step) {
+      case ChallengeTeamWizardStep.SELECT_YOUR_CHALLENGE_TEAM:
+      case ChallengeTeamWizardStep.JOIN_REQUEST_SENT:
+      case ChallengeTeamWizardStep.REGISTRATION_SUCCESSFUL:
+        // Cannot go back from these steps
+        return <></>
+      case ChallengeTeamWizardStep.ACCEPT_INVITATION:
+      case ChallengeTeamWizardStep.JOIN_REQUEST_FORM:
+      case ChallengeTeamWizardStep.CREATE_NEW_TEAM:
+      default:
+        // All the other steps go back to team selection
+        return (
+          <Button
+            variant={'outlined'}
+            onClick={() => {
+              setStep(ChallengeTeamWizardStep.SELECT_YOUR_CHALLENGE_TEAM)
+              setSelectedTeamId(undefined)
+            }}
+          >
+            Back
+          </Button>
         )
-        setHasSubmissionTeam(isReg)
-      }
-      setLoading(false)
-    }
-    if (userSubmissionTeamError) {
-      setErrorMessage(
-        `Error: Could not determine if you are already registered for this Challenge.`,
-      )
-      setLoading(false)
     }
-  }, [userSubmissionTeams, userSubmissionTeamError])
-
-  const { data: teamMembershipStatus, error: teamMembershipError } =
-    useGetMembershipStatus(
-      selectedTeamId ?? EMPTY_ID,
-      userProfile?.ownerId ?? EMPTY_ID,
-      {
-        enabled: !!selectedTeamId && !!userProfile,
-      },
+  }, [step])
+
+  const closeButton = useMemo(() => {
+    return (
+      <Button
+        variant={'contained'}
+        onClick={() => {
+          onClose()
+        }}
+      >
+        Close
+      </Button>
     )
+  }, [onClose])
 
-  useEffect(() => {
-    if (teamMembershipStatus) {
-      setMembershipStatus(prev => ({
-        ...prev,
-        [teamMembershipStatus.teamId]: teamMembershipStatus,
-      }))
+  // Determine modal content based on step.id
+  const { actions = <></>, content = <></> } = useMemo(() => {
+    if (!isLoggedIn) {
```

Comment:
Instead of using an effect to `setErrorMessage` when not logged in, or when already on a submission team, just return the error in an `Alert`.